app/services/notification_service.rb

```diff
@@ -24,19 +25,26 @@ def parse_phone_number(number)
     default_country_code + parsed_number
   end
 
+  def failed?
+    error.present?
+  end
+
   private
 
   def default_country_code
     Rails.application.config.country[:sms_country_code]
   end
 
   def send_twilio_message(sender_number, recipient_number, message, callback_url = nil)
-    client.messages.create(
+    @response = client.messages.create(
       from: sender_number,
       to: recipient_number,
       status_callback: callback_url,
       body: message
     )
+  rescue Twilio::REST::TwilioError => e
+    report_error(e)
+    @error = :exception
```

Comment:
How about

```
rescue Twilio::REST::TwilioError => exception
  @exception = exception
  report_error(exception)
end
```

We may as well capture the actual exception here rather than a symbol.

Reply:
WIll do.